artemis/modules/classifier.py

```diff
@@ -119,37 +133,78 @@ def run(self, current_task: Task) -> None:
 
         sanitized = self._sanitize(data)
         task_type = self._classify(sanitized)
+        self.db.save_task_result(task=current_task, status=TaskStatus.OK, data={"type": task_type, "data": [sanitized]})
+
+        if task_type == TaskType.SERVICE:
+            host, port_str = data.split(":")
+            port = int(port_str)
+
+            if is_domain(host):
+                host_type = "domain"
+            else:
+                host_type = "ip"
+
+            try:
+                output = throttle_request(
+                    lambda: check_output_log_on_error(
+                        ["fingerprintx", "--json"], self.log, input=data.encode("ascii")
+                    ).strip()
+                )
+            except subprocess.CalledProcessError:
+                self.log.exception("Unable to fingerprint %s", data)
+                return
+
+            if not output:
+                self.log.exception("Unable to fingerpritn %s", data)
+                return
+
+            data = json.loads(output)
+            ssl = data["tls"]
+            service = data["protocol"]
```

Comment:
will output of fingerprintx have these fields every time? looking at it's docs - I cannot really tell